views.py

```diff
@@ -73,9 +76,46 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
     else:
         res_root = res_id
 
-    bag_modified = "false"
+    if not istorage.exists(res_root):
+        return HttpResponseBadRequest("ResourceId {res_id} does not exist".format(res_id=res_id));
```

Comment:
@sblack-usu It'd work better if you move this ```istorage.exists()``` call to before ```authorize()``` call so that the bad request response can be immediately returned if the pass in resource id is not valid, before doing any further authorization check.